### PR TITLE
docs: update local development docs

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Update development docs
- Update versions of tools/programs where needed
- Capitalize proper noun `Docker`
- Rewrite some sentences for simplicity

## Context

Drive-by PR, not closing any issue with this.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/local-development.md [hunk 1]
```diff
@@ -132,8 +132,8 @@ You are better off to instead export the Environment Variable `RENOVATE_TOKEN` f
 
 ### Run against a real repo
 
-To make sure everything is working, create a test repo in your account, e.g. like `https://github.com/r4harry/testrepo1`.
-Now, add a file called `.nvmrc` with the content `8.13.0`.
+To make sure everything is working, create a test repository in your account, e.g. like `https://github.com/r4harry/testrepo1`.
+Now, add a file called `.nvmrc` with the content `20.12.2`.
```
Comment (HonkingGoose) [inline docs/development/local-development.md:136]: The `20.12.2` version is the current version in our own `.npmrc` file. So we should probably use that number?

https://github.com/renovatebot/renovate/blob/e3351ba4c64d3169b783471245a5e5ce06d28feb/.nvmrc#L1C1-L1C8

File: docs/development/local-development.md [hunk 2]
```diff
@@ -10,9 +10,9 @@ For example, if you think anything is unclear, or you think something needs to b
 
 You need the following dependencies for local development:
 
-- Git `>=2.33.0`
+- Git `>=2.45.1`
```
Comment (HonkingGoose) [inline docs/development/local-development.md:13]: Git recently released `2.45.1` to fix several CVEs, so might be good to bump our recommendation.

In any case, Renovate itself uses at least `2.45.1`.

File: docs/development/local-development.md [hunk 3]
```diff
@@ -10,9 +10,9 @@ For example, if you think anything is unclear, or you think something needs to b
...
 - Node.js `^18.12.0 || >=20.0.0`
-- pnpm `^8.6.11` (use corepack)
+- pnpm `^9.0.0` (use corepack)
```
Comment (HonkingGoose) [inline docs/development/local-development.md:15]: This new number for pnpm comes from our `package.json`'s `engine` field:

https://github.com/renovatebot/renovate/blob/e3351ba4c64d3169b783471245a5e5ce06d28feb/package.json#L136-L139

**Review by HonkingGoose — COMMENTED** [review]
Commenting to explain where I'm getting the new numbers from.

File: docs/development/local-development.md (hunk 1 above)
Comment (rarkins) [reply docs/development/local-development.md:136]: The reason for nvmrc was to have a very simple package file example. It doesn't need to be the latest

Comment (rarkins) [reply docs/development/local-development.md:136]: 20.0.0 can be sufficient

Comment (HonkingGoose) [reply docs/development/local-development.md:136]:
```suggestion
Now, add a file called `.nvmrc` with the content `20.0.0`.
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**